src/Symfony/Component/Cache/DoctrineProvider.php

```diff
@@ -15,106 +15,108 @@
 use Psr\Cache\CacheItemPoolInterface;
 use Symfony\Contracts\Service\ResetInterface;
 
-/**
- * @author Nicolas Grekas <p@tchwork.com>
- *
- * @deprecated Use Doctrine\Common\Cache\Psr6\DoctrineProvider instead
- */
-class DoctrineProvider extends CacheProvider implements PruneableInterface, ResettableInterface
-{
-    private $pool;
-
-    public function __construct(CacheItemPoolInterface $pool)
-    {
-        trigger_deprecation('symfony/cache', '5.4', '"%s" is deprecated, use "Doctrine\Common\Cache\Psr6\DoctrineProvider" instead.', __CLASS__);
-
-        $this->pool = $pool;
-    }
-
+if (class_exists(CacheProvider::class)) {
```

Comment:
We should be able to use en early return statement and save reindenting the file, isn't it